src/platform/packages/shared/kbn-esql-utils/src/utils/replace_index_pattern.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { ESQLSource, EsqlQuery } from '@kbn/esql-ast';
+
+export function replaceESQLQueryIndexPattern(esql: string, replacements: Record<string, string>) {
+  const query = EsqlQuery.fromSrc(esql);
+  const sourceCommand = query.ast.commands.find(({ name }) => ['from', 'ts'].includes(name));
```

Comment:
what is the `ts` command?

Reply:
see https://github.com/elastic/kibana/pull/217288#discussion_r2042580241